### PR TITLE
fix dev docs search results priority;

### PR DESCRIPTION
**Pull request by a-hariti**
please rename `DEVELOPER_DOCS=1` to `NEXT_PUBLIC_DEVELOPER_DOCS=1` before the merge

both on Github secrets and on Vercel

### CODE REVIEW

**Review by chargome — CHANGES_REQUESTED** [review]
@a-hariti Thank you! We should also update the env var in our algolia github action.

**Comment by a-hariti** [comment]
Yes, that's what I said on the PR description

**Comment by chargome** [comment]
I meant the actual workflow: `algolia-index.yml`

**Review by chargome — APPROVED** [review]

**Comment by chargome** [comment]
Note: I have added the env var in vercel, still need to confirm the github secret update by @stephanie-anderson

File: src/middleware.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import type {NextRequest} from 'next/server';
 import {NextResponse} from 'next/server';
 
-// This env var is set in next.config.js based on the `DEVELOPER_DOCS` env var at build time
+// This env var is set in next.config.js based on the `NEXT_PUBLIC_DEVELOPER_DOCS` env var at build time
 // a workaround edge middleware not having access to env vars
 const isDeveloperDocs = process.env.DEVELOPER_DOCS_;
```
Comment (stephanie-anderson) [inline src/middleware.ts:6]: @a-hariti is the underscore intentional?

Comment (stephanie-anderson) [reply src/middleware.ts:6]: And should it also be renamed?

Comment (chargome) [reply src/middleware.ts:6]: @stephanie-anderson this is intentional since this env var is being set in `next.config.js`

Comment (a-hariti) [reply src/middleware.ts:6]: yes, it is

That's a workaround to avoid conflict with the actual env var without an `_`

**Pull request merged**